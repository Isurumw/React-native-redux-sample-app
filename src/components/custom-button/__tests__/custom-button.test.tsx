import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';

import CustomButton from '../custom-button';

const mockButtonPress = jest.fn();

describe('render the custom button', () => {
  beforeEach(() => {
    mockButtonPress.mockClear();
  });

  it('the button component renders correctly', () => {
    const component = render(
      <CustomButton label="SIGN IN" onPress={mockButtonPress} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('tapped the button', () => {
    const component = render(
      <CustomButton label="SIGN IN" onPress={mockButtonPress} />,
    );

    const button = component.getByTestId('test_button');
    fireEvent.press(button);

    expect(mockButtonPress).toBeCalledTimes(1);
  });
});
