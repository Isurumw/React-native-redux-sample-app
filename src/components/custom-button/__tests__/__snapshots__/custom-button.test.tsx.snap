// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the custom button the button component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#7C2AF7",
      "borderRadius": 20,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="test_button"
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 13,
        "letterSpacing": 0.02,
        "lineHeight": 16,
        "marginStart": 14,
      }
    }
  >
    SIGN IN
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/chevron_right.png",
      }
    }
    style={
      Object {
        "height": 15,
        "width": 15,
      }
    }
  />
</View>
`;
